### PR TITLE
[VideoPlayer] Adjust lastseekpts on FlushBuffers

### LINKED ISSUE #24324: Jump / seek makes video to freeze for few seconds if playback speed is increased
### Describe the bug
Here is a clear and concise description of what the problem is:

(Note: this is about `tempoup` and not about "fast forward".)

For me video always freezes for few seconds even if I jump just for 10 seconds ahead.
I tried to increase cache - it did not help. I mean for local video file, not even a network stream. I also tried to disable hardware decoding, switching other video related settings - no effect.
And I have this issue on all my devices / OSes: certified android tv box with Amlogic S905X4 and PC with Linux/Windows with i5-8259U. For Windows it was freshly installed kodi with default settings.
I tested on videos such: 720p h264, 1080p h264, 4k h265.
Just to be sure that this is not hardware issue: there is no freeze on Linux with vlc / mpv and Android TV with mx player / vimu.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Jump / seek should not make video to freeze if playback speed is increased.

## Actual Behavior

Jump / seek makes video to freeze for few seconds if playback speed is increased.

### To Reproduce

1. open video file
2. increase speed (1.5 fox example) via `tempoup` (not via "fast forward")
3. do "jump ahead" for 10 seconds (e.g. press right arrow on keyboard)
4. see how video freezes for few seconds

### Debuglog

The debuglog can be found here:
https://paste.kodi.tv/uyiyorisox.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Linux 6.5.0-14 (Ubuntu 23.10), Android (Google) TV 11, Windows 11
 - Kodi version: 20.2

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description
If you ff or rw video and execute a seek afterwards, videoplayer gets stuck doing recursive seeks to adjust the clock. Since the pts of the seek is not saved, the difference between the clock and the new position is higher than the threshold forcing yet another seek (https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L2179-L2193). This can happen multiple times, stalling the video in such situations.
With this change seeks are pretty much instantaneous. Furthermore, the behaviour of the ff with higher rates (16x 32x) is also improved and more fluid.

@FernetMenta hopefully that's what you meant with the cosmetic suggestions in https://github.com/xbmc/xbmc/issues/24324#issuecomment-1935408613
Also took the time to remove the memsets and nuke the legacy comment we discussed previously.

## Motivation and context
Fix https://github.com/xbmc/xbmc/issues/24324

## How has this been tested?
Seeking while ff/rw, checking player doesn't stall. ff/rw at high rates.
Tested tempo as well.

## What is the effect on users?
ff and rw should work way better now and not be stalled with seeks.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayer.h [hunk 1]
```diff
@@ -514,10 +514,19 @@ class CVideoPlayer : public IPlayer, public CThread, public IVideoPlayer,
...
-    double lastpts;  // holds last display pts during ff/rw operations
-    int64_t lasttime;
-    double lastseekpts;
-    double lastabstime;
+    double lastpts{0.0}; // holds last display pts during ff/rw operations
+    int64_t lasttime{0};
+    double lastseekpts{0.0};
+    double lastabstime{0.0};
+
+    void FlushBuffers(double pts)
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayer.h:522]: FlushBuffers might be mislieading as a name here bacause this struct does not have any buffers. Maybe just "reset"
Apart from that it looks good imo.

Comment (enen92) [reply xbmc/cores/VideoPlayer/VideoPlayer.h]: Might have been mislead by your comment "reset method (with current pts) to this struct **and call it FlushBuffers too.** ". I agree Reset is better, done on the last push.
Thanks for your feedback and availability

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoPlayer.h]: sorry, I forgot the "in"

**Comment by CrystalP** [comment]
That's a noticeable improvement, thanks.
I was hoping it would also improve seeking with tempo play but it doesn't (no picture/audio for a few seconds after seek)
Is there maybe another place that needs a similar fix? (I'm really not familiar with the player)

**Pull request merged**